### PR TITLE
ROX-10215 Update Dashboard to use URL Search params instead of redux

### PR DESCRIPTION
**Pull request by dvail**
## Description

This updates the Dashboard page to use URL Parameters to store search state instead of Redux. User facing functionality should be nearly identical except for the following changes:
- Search parameters are reflected in the URL of the main dashboard page when the search filter is changed
- Links from the charts to the Violations page now correctly apply search filters
- Links from the pie charts also apply a "Cluster" filter on the Violations page*

(* This was not the behavior of the old, broken links, but is consistent with the way links are handled from the bar chart. This also seems like more consistent UX since the pie charts are affected by the cluster filter, so it seems like the violations shown on the Violations page should match the data the user is coming from.)

There are a handful of follow up PRs that I'd like to make separate after this one is merged.
- Fix "System violations" tile links and associated test #1366
- Clean up now dead code from Redux/reducers/sagas #1367
- Clean up handling of `useEffect` to avoid memory leaks pending team discussion on how best to approach this
- Convert these files to TS if time allows

## Checklist
- [ ] Investigated and inspected CI test results

## Testing Performed

Test that all cluster data loads on initial page visit:
image

Test that search filter correctly filters data for overall violation counts and charts:
Cluster: production
image
image

Cluster: security
image
image

Test that links from bar graph correctly apply filters on Violations page
image

Test that links from pie graph correctly apply filters on Violations page
image

Test that "Top Risky Deployments" widget matches the deployment order on the "Risk" page
image
image

### CODE REVIEW

File: ui/apps/platform/src/Components/SearchFilterInput/SearchFilterInput.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ type SearchFilterInputProps = {
     className: string;
     handleChangeSearchFilter: (searchFilter: SearchFilter) => void;
     placeholder: string;
-    searchCategory: SearchCategory;
+    searchCategory?: SearchCategory;
```
Comment (dvail) [inline ui/apps/platform/src/Components/SearchFilterInput/SearchFilterInput.tsx:12]: The backend API supports requests without a search category, so I've updated this prop to be optional.

File: ui/apps/platform/src/services/AlertsService.ts [hunk 2]
```diff
@@ -77,17 +75,13 @@ type SummaryAlertCountsFilters = {
 /*
  * Fetch severity counts.
  */
-export function fetchSummaryAlertCounts(
-    filters: SummaryAlertCountsFilters
-): Promise<{ response: { groups: AlertGroup[] } }> {
+export function fetchSummaryAlertCounts(filters: SummaryAlertCountsFilters): Promise<AlertGroup[]> {
```
Comment (dvail) [inline ui/apps/platform/src/services/AlertsService.ts:78]: These were only used from the main dashboard page, so I updated the type signatures to remove some extra wrapping and unwrapping of the return values.

**Comment by dvail** [comment]
Failing e2e tests... 👀

**Review by sachaudh — APPROVED** [review]
LGTM. Thanks for making the changes. I was going to comment about removing unused reducers, but you mentioned you have separate PRs for those so that's great 👍🏼

**Pull request merged**